### PR TITLE
Update boost 1.74

### PR DESCRIPTION
**Pull request by o01eg**

### CODE REVIEW

**Comment by geoffthemedio** [comment]
What did you change to get it to build?

**Comment by o01eg** [comment]
I suppose environment variables was the reason of build error. Other step didn't help.

File: .travis.yml [hunk 1]
```diff
@@ -55,28 +55,44 @@ jobs:
...
       <<: *_build_sdk_common
       os: windows
+      compiler: cl
       env:
         - RELEASES_FILE=FreeOrionSDK_*.zip
         - RELEASES_TAG_NAME=$TRAVIS_TAG
+        - CC=cl.exe
+        - CXX=cl.exe
+        - CC_FOR_BUILD=cl.exe
+        - CXX_FOR_BUILD=cl.exe
```
Comment (geoffthemedio) [inline .travis.yml:93]: Are these the relevant environment variables...?

Comment (o01eg) [reply .travis.yml:93]: Travis will set them all to gcc/g++ if not set so I think it better to set all of them.

**Comment by geoffthemedio** [comment]
SDK builds OK on windows in MSVC 2017. In 2019 there are GLEW linker errors, probably unrelated to Boost.

I get some errors building FreeOrion itself, but can resolve those, I think.

**Comment by o01eg** [comment]
@geoffthemedio Should I clean up other changes to merge this PR?

**Comment by geoffthemedio** [comment]
what needs cleaning up?

**Comment by o01eg** [comment]
MacOS changes I made to pass CI and removal of mingw?

**Comment by geoffthemedio** [comment]
sure, go ahead...

**Comment by o01eg** [comment]
@geoffthemedio Updated. Not sure what will happened with MacOS.

**Comment by geoffthemedio** [comment]
@Vezzra previously reported it worked, so presumably it will be OK

**Comment by geoffthemedio** [comment]
Maybr re-push to restart the TravisCI build? Not sure if the failure is legitimate...

**Comment by o01eg** [comment]
That is why I disabled homebrew update.

**Comment by geoffthemedio** [comment]
> 
> 
> That is why I disabled homebrew update.

Have you re-unabled it? The MacOS build is still failing and the build log contains
````
$ brew install gpatch
Updating Homebrew...
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs
````

**Comment by o01eg** [comment]
@geoffthemedio No, update was enabled in `master`.

**Pull request merged**